### PR TITLE
Extract Swapbox to a component

### PR DESCRIPTION
**Pull request by Diogomartf**
# Summary

Swapbox should be it's own component, so it's self contained and removes logic from swap index (landing page).

image

# To Test

1. Go to swap page
- [ ] test making swaps in diferent networks
- [ ] test swapbox toggles
- [ ] test swapbox modals
- [ ] Is expected that swapbox looks and behaves the same

### CODE REVIEW

File: src/pages/Swap/index.tsx
Comment (mwiniarsk) [inline src/pages/Swap/index.tsx:134]: If we are cleaning swap page I would also move ButtonGroup to a separate file

Comment (Diogomartf) [reply src/pages/Swap/index.tsx:134]: Yes, that's a nice suggestion.

Comment (Diogomartf) [reply src/pages/Swap/index.tsx]: Done.

**Review by mwiniarsk — APPROVED** [review]

File: src/pages/Swap/index.tsx
Comment (Diogomartf) [reply src/pages/Swap/index.tsx:114]: We are only using it here, do you think it's worth it?

Comment (berteotti) [reply src/pages/Swap/index.tsx:114]: Since it's not being used elsewhere in `Swap/index` I think we can extract. Code will be cleaner in this component. But it's just a small improvement.

Comment (Diogomartf) [reply src/pages/Swap/index.tsx:114]: Ok, fair, I'll do that.

**Comment by MilanVojnovic95** [comment]
Wrong display of error message when transaction is rejected. 
image

it should look like this:
image

the rest looks ok. 👍

**Comment by Diogomartf** [comment]
> Wrong display of error message when transaction is rejected. image
> 
> it should look like this: image
> 
> the rest looks ok. 👍

I think you are in "expert mode", so that's how the error looks. In normal mode shows the modal.

**Comment by Mi-Lan** [comment]
@Diogomartf Hey wondering why is this pr based on simple chart?

**Comment by Diogomartf** [comment]
> @Diogomartf Nice change definitely declutters the code abit! I would just add that I think Swapbox shouldn't be in components folder yet...Since it contains tons of logic as well. For now I think its fine to leave it inside the pages folder. And once we decide to decouple logic we can slowly start moving it to components folder....

Yes, it makes sense. I'll do that.

**Comment by Diogomartf** [comment]
> @Diogomartf Hey wondering why is this pr based on simple chart?

I was working on simple chart when I noticed that we should extract Swapbox to its own component, so I've made the change on top of that.  Probably should have done it directly on develop. 

But I think this change can go with simple chart, what do you think?

**Comment by Mi-Lan** [comment]
> > @Diogomartf Hey wondering why is this pr based on simple chart?
> 
> I was working on simple chart when I noticed that we should extract Swapbox to its own component, so I've made the change on top of that. Probably should have done it directly on develop.
> 
> But I think this change can go with simple chart, what do you think?

I think for the sake of making code review on simple chart easier we wait until its merged then merge this after...

**Comment by Diogomartf** [comment]
> > > @Diogomartf Hey wondering why is this pr based on simple chart?
> > 
> > 
> > I was working on simple chart when I noticed that we should extract Swapbox to its own component, so I've made the change on top of that. Probably should have done it directly on develop.
> > But I think this change can go with simple chart, what do you think?
> 
> I think for the sake of making code review on simple chart easier we wait until its merged then merge this after...

Agree. Let's merge it after simple chart.

**Review by mix1o — APPROVED** [review]

**Comment by berteotti** [comment]
Can we fix this in this PR?

image

**Review by berteotti — APPROVED** [review]
Looks good to me

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Its def good idea to declutter the swapBox. But here are couple of suggestions if we wanna refactor the swapbox:
1) Create container that will be in pages and rest of UI components move to components folder so we decouple logic from styling 
2) Implement storybook for swapbox so we make sure its really decoupled
3) Folder/File strucutre=> ChartOptionButton and ChartToogle seem abit out of place I think they belong to components folder since they are mostly presentation components

**Comment by Diogomartf** [comment]
> Its def good idea to declutter the swapBox. But here are couple of suggestions for improving the swapbox:
> 
> 1. Create container that will be in pages and rest of UI components move to components folder so we decouple logic from styling
> 2. Implement storybook for swapbox so we make sure its really decoupled
> 3. Folder/File strucutre=> ChartOptionButton and ChartToogle seem abit out of place I think they belong to components folder since they are mostly presentational components

Thanks for the feedback.

I think it will be too much work to remove logic from the `swapbox` right now. This PR is a step forward on that direction. If we want to do it we have to plan it. This is an incremental improvement. There is also a plan to change the `swapbox` design with the new token picker so I would do it after this. 

The advantage of having it on storybook is if we were able to reuse it on other places. Which now I think it's not the case, there is some potential similarities with `BridgeBox` but this requires design thinking and a refactorings.

About `ChartOptionButton`, I forgot to delete, it was an old implementation of `ChartToggle`. Deleted now. Thanks!

About `ChartToogle` is coupled with `swap index` that's why I think they are better near the `swapbox`. They are using components like `ButtonGroup` (it is on storybook). I just extracted to it's own component to declutter the main page.

I think we can merge this PR as it.

**Comment by Mi-Lan** [comment]
> > Its def good idea to declutter the swapBox. But here are couple of suggestions for improving the swapbox:
> > 
> > 1. Create container that will be in pages and rest of UI components move to components folder so we decouple logic from styling
> > 2. Implement storybook for swapbox so we make sure its really decoupled
> > 3. Folder/File strucutre=> ChartOptionButton and ChartToogle seem abit out of place I think they belong to components folder since they are mostly presentational components
> 
> Thanks for the feedback @Mi-Lan 🙌
> 
> I think it will be too much work to remove logic from the `swapbox` right now. This PR is a step forward on that direction. If we want to do it we have to plan it. This is an incremental improvement. There is also a plan to change the `swapbox` design with the new token picker so I would do it after this.
> 
> The advantage of having it on storybook is if we were able to reuse it on other places. Which now I think it's not the case, there is some potential similarities with `BridgeBox` but this requires design thinking and refactorings.
> 
> About `ChartOptionButton`, I forgot to delete, it was an old implementation of `ChartToggle`. Deleted now. Thanks!
> 
> About `ChartToogle` is coupled with `swap index` that's why I think it is better to keep it near the `swap index (page)`. `CharToogle` is using components like `ButtonGroup` (it is on storybook). It's a concrete implementation using reusable components, I just extracted to it's own component to declutter the main page.
> 
> There are a lot of interesting ideas we can continue to work on in another PRs. I think we can merge this PR as it for now.

Yeah I agree It would be tons of work to do this now in this pr disregard that.

So the advantage of having components in storybook is not to have them just more easily reused... but also to style components within it thats the main use case in general.

I think that should still be moved to components just because its only used there doesn't justify it being there when its a presentational component we should follow the rules rest of components follow and move it. Theres other components that are also used just once...

**Comment by Diogomartf** [comment]
> So the advantage of having components in storybook is not to have them just more easily reused... but also to style components and develop them within storybook thats the main use case in general.

Ok, got you on Storybook, makes sense. We might want to do that in the future.

I still think that would be better to have it close but to be consistent with our codebase I'll move it to components folder.

**Comment by berteotti** [comment]
@Diogomartf @Mi-Lan just to let you know components are being moved into the page folder in limit orders PR https://github.com/levelkdev/swapr-dapp/pull/1356/files 

I think this is better so we don't have a large component folder with components being used only in 1 page.

So I think it's better to have Swapbox next to Swap page for now. And move it to components when it's logic is decoupled from the UI.

**Comment by Mi-Lan** [comment]
> @Diogomartf @Mi-Lan just to let you know, components are being moved into the page folder in limit orders PR #1356
> 
> I think this is better so we don't have a large component folder with components being used only in 1 page.
> 
> So I think it's better to have Swapbox next to Swap page for now. And move it to components when it's logic is decoupled from the UI.

Yeah I agree for swapbox but was reffering to that chartToogle component

**Review by Mi-Lan — APPROVED** [review]
Lets go!

**Pull request merged**
